internal/translation/privateendpoint/conversion.go

```diff
@@ -473,3 +473,30 @@ func interfaceCreateToAtlas(peInterface EndpointInterface, gcpProjectID string)
 
 	return nil
 }
+
+type CompositeEndpointInterface struct {
```

Comment:
This name is too general, the composited types are concrete while the name is abstract. Just call the type very concrete, i.e. "Endpoints", see https://github.com/mongodb/mongodb-atlas-kubernetes/blob/46a504c6fa326037607bcc51e1518c3ede092330/internal/translation/datafederation/conversion_endpoints.go#L69-L71

Reply:
@s-urbaniak the interface suffix, given the Go reserved word, is probably making you think this is an abstraction, but it's not. They are, indeed, the interface part of the private endpoint, that's why it's named like that.
With this, are you okay keeping the name or prefer to keep your suggestion?
Note: The example from data federation differ from here because the interface there is predefined and not a concept in the API

Reply:
lol, indeed, resolving